From 26e8623e687e30fa9195e9da22479ccdb6afc91b Mon Sep 17 00:00:00 2001
From: Artem Lapkin <art@khadas.com>
Date: Wed, 10 Nov 2021 13:22:16 +0800
Subject: [PATCH] arm64: dts: rockchip: emmc remove mmc-hs400-enhanced-strobe
 rk3399-khadas-edge

Remove mmc-hs400-enhanced-strobe for rk3399-khadas-edge

Not compitable for some eMMC chips ( BJTD4R 29.1 GiB ). And no
performance difference in our case from emmc-hs400-enhanced-strobe option
for other eMMC chips.

Problem example ( eMMC chip BJTD4R 29.1 GiB ):

[    0.068282] platform ff770000.syscon:phy@f780: Fixing up cyclic dependency with fe330000.mmc
[    7.001493] mmc2: CQHCI version 5.10
[    7.027971] mmc2: SDHCI controller on fe330000.mmc [fe330000.mmc] using ADMA
.......
[    7.207086] mmc2: mmc_select_hs400es failed, error -110
[    7.207129] mmc2: error -110 whilst initialising MMC card
[    7.308893] mmc2: mmc_select_hs400es failed, error -110
[    7.308921] mmc2: error -110 whilst initialising MMC card
[    7.427524] mmc2: mmc_select_hs400es failed, error -110
[    7.427546] mmc2: error -110 whilst initialising MMC card
[    7.590993] mmc2: mmc_select_hs400es failed, error -110
[    7.591012] mmc2: error -110 whilst initialising MMC card

Workable example without mmc-hs400-enhanced-strobe:

[    6.960785] mmc2: CQHCI version 5.10
[    6.984672] mmc2: SDHCI controller on fe330000.mmc [fe330000.mmc] using ADMA
[    7.175021] mmc2: Command Queue Engine enabled
[    7.175053] mmc2: new HS400 MMC card at address 0001
[    7.175808] mmcblk2: mmc2:0001 BJTD4R 29.1 GiB 
[    7.176033] mmcblk2boot0: mmc2:0001 BJTD4R 4.00 MiB 
[    7.176245] mmcblk2boot1: mmc2:0001 BJTD4R 4.00 MiB 
[    7.176495] mmcblk2rpmb: mmc2:0001 BJTD4R 4.00 MiB, chardev (242:0)

Performance note for mmc-hs400-enhanced-strobe usage:

Other chips DUTA42 116 GiB works well with or without
mmc-hs400-enhanced-strobe!

..... mmc-hs400-enhanced-strobe disabled

786432000 bytes (786 MB, 750 MiB) copied, 3 s, 262 MB/s

..... mmc-hs400-enhanced-strobe enabled

[    7.135880] mmc2: Command Queue Engine enabled
[    7.135928] mmc2: new HS400 Enhanced strobe MMC card at address 0001
[    7.136992] mmcblk2: mmc2:0001 DUTA42 116 GiB 

1048576000 bytes (1.0 GB, 1000 MiB) copied, 4 s, 262 MB/s

Signed-off-by: Artem Lapkin <art@khadas.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-khadas-edge.dtsi | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-khadas-edge.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-khadas-edge.dtsi
index d5c7648c841d..f1fcc6b5b402 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-khadas-edge.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-khadas-edge.dtsi
@@ -705,7 +705,6 @@ &sdmmc {
 &sdhci {
 	bus-width = <8>;
 	mmc-hs400-1_8v;
-	mmc-hs400-enhanced-strobe;
 	non-removable;
 	status = "okay";
 };
-- 
2.25.1

